docs/personal_registry.md

```diff
@@ -1,27 +1,111 @@
 ---
-title: Setting a Personal Registry with a shared Tiltfile
-description: "Tilt's 'default_registry' function lets you change the image repository so you don't have to ask permission to get started."
+title: Choosing an Image Registry
+description: "A guide to dev image registries and how to get your code into a cluster."
 layout: docs
 sidebar: guides
 ---
 
-You should be able to start an app on Kubernetes from just the source code. One stumbling block is when YAML and scripts hard-code the image repository (the `gcr.io/windmill` in `gcr.io/windmill/user-service`). Tilt's `default_registry` function lets you change the image repository so you don't have to ask permission to get started. This guide walks you through two cases:
-* for users, how to experiment with an existing Tilt project by adding one line to the Tiltfile
-* for maintainers, how to configure your project so users don't need to modify the Tiltfile at all
+If you're building container images in dev, you'll need a place to put those
+images where your cluster can pull them.
 
-## Experiment with an existing Tilt project
-You've cloned a repo, run `tilt up`, and see push errors saying you lack permission. Pick a registry that you can push to, like `gcr.io/my-personal-project`. Add the following line to the `Tiltfile`:
+## The Easy Way (for 95% of Users)
 
-```python
-default_registry('gcr.io/my-personal-project')
+Here's how you set up a dev image registry.
+
+Step 1) [Choose a cluster](choosing_clusters.html).
+
+Step 2) There is no step 2!!!
+
+### Your Registry is Usually Already Set Up For You
+
+For almost all clusters, the cluster will have a registry for you.
+
+- If you're using Docker for Desktop, there's no registry at all. You build
+  directly into the container runtime.
+
+- If you're using Kind, [our setup scripts](choosing_clusters.html) will set up
+  a registry for you.
+
+- If you're using a remote cluster (like AKS, EKS, GKE, and DigitalOcean
+  Kuberentes), you'll have a registry that's colocated with your cluster.
+
+### Your Registry Secrets are Usually Already Set Up For You
+
+You also don't have to worry about authentication.
+
+Our local cluster guides set up registries without any auth (because they're purely local).
+
+If you're using an authenticated registry, that's usually not a problem either.
+Tilt never talks directly to a registry. When Tilt builds images, it tells the
+image builder to push to the registry when it's done. If you need to login to a
+registry, you'll login with the image builder.
+
+For Docker Hub, you run:
+
+```
+docker login
 ```
```

Comment:
Suggest to add a link to the docs for `docker login` in case the user runs into any issues
https://docs.docker.com/engine/reference/commandline/login/

Reply:
:+1: